lib/sync.js

```diff
@@ -466,7 +469,7 @@ SyncApi.prototype._sync = function(syncOptions, attempt) {
                 var accountDataEvents = self._mapSyncEventsFormat(joinObj.account_data);
 
                 // we do this first so it's correct when any of the events fire
-                room.unread_notification_count = joinObj.unread_notification_count;
+                room.unread_notifications = joinObj.unread_notifications || {};
```

Comment:
I was okay when this was just a number, but now this is representing the same structure as the /sync response. One of the things which has saved our figurative bacon in the past is the shielding the JS SDK provides against breaking API changes (which made the v2 swap that much easier).

Please can you add a setter/getter for this on the `Room` object, e.g:

```javascript
Room.setUnreadNotifications(highlightCount, notificationCount)
Room.getNotificationCount(isHighlight)
```